File: src/main/java/com/example/agentdeepseek/model/entity/AgentConfig.java
package com.example.agentdeepseek.model.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * Agent配置实体类
 * 存储自定义Agent的完整配置信息
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AgentConfig {
    private Long id;
    private String name;
    private String description;
    private String avatar;
    private String systemPrompt;
    private String toolNames;
    private String modelName;
    private String thinkingMode;
    private String executionMode;
    private Double temperature;
    private String workDir;
    private Integer sortOrder;
    private Integer enabled;
    private Integer isDefault;
    private Integer isBuiltin;
    /** 绑定的 LLM Provider ID，关联 llm_provider.id（null 或 0 表示使用默认 Provider） */
    private Long providerId;
    /** 前端动态选择的 LLM Provider Code（如 deepseek/openai），运行时持久化 */
    private String providerCode;
    /** 角色性格配置 JSON 字符串（如 {"name":"圆圆","species":"人",...}） */
    private String characterProfile;
    private Long userId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/api/agent-config.ts
import { request } from '@/utils/http-client'

export interface AgentConfig {
  id?: number
  name: string
  description?: string
  avatar?: string
  systemPrompt?: string
  toolNames?: string
  tools?: string[]
  modelName?: string
  thinkingMode?: string
  executionMode?: string
  workDir?: string
  sortOrder?: number
  enabled?: number
  isDefault?: number
  isBuiltin?: number
  userId?: number
  createdAt?: string
  updatedAt?: string
}

/**
 * 获取用户可用的 Agent 列表
 */
export async function listAgentConfigs() {
  return request<AgentConfig[]>('/agent-configs/list')
}

/**
 * 创建 Agent
 */
export async function createAgentConfig(data: AgentConfig) {
  return request<AgentConfig>('/agent-configs/create', {
    method: 'POST',
    body: JSON.stringify(data)
  })
}

/**
 * 更新 Agent
 */
export async function updateAgentConfig(id: number, data: AgentConfig) {
  return request<void>(`/agent-configs/${id}`, {
    method: 'PUT',
    body: JSON.stringify(data)
  })
}

/**
 * 删除 Agent
 */
export async function deleteAgentConfig(id: number) {
  return request<void>(`/agent-configs/${id}`, { method: 'DELETE' })
}

/**
 * 设为默认 Agent
 */
export async function setDefaultAgentConfig(id: number) {
  return request<void>(`/agent-configs/${id}/default`, { method: 'PUT' })
}

/**
 * 更新 Agent 运行时配置（模型、思考模式、执行模式、工作目录）
 */
export async function updateAgentRuntime(id: number, data: Partial<AgentConfig>) {
  return request<AgentConfig>(`/agent-configs/${id}/runtime`, {
    method: 'PUT',
    body: JSON.stringify(data)
  })
}
